frontend/src/frontend/code_generator/user_callgraph.py

```diff
@@ -143,10 +143,60 @@ def format_branch(self, branch: blocks.Branch) -> str:
             return branch_formatters[branch.branch_type](branch)
         raise ValueError(f'Unknown branch type: {branch.branch_type}')
 
-    def _format_branch_indirect_call(self, branch: blocks.Branch) -> str:
+    def _format_branch_indirect_call(self,
+                                     branch: blocks.Branch,
+                                     uuid: int = None) -> str:
+        if uuid is None:
+            uuid = id(branch)
+        if len(branch.get_targets()) == 1:
+            return self._format_indirect_call_singletarget(branch, uuid)
+        return self._format_indirect_call_multitarget(branch, uuid)
+
+    def _format_indirect_call_singletarget(self, branch: blocks.Branch,
+                                           uuid: int) -> str:
         target = branch.next_valid_target()
         sig = self.function_call_signature_for(target)
-        return f'void (*frontend_f)(void) = {sig};\nfrontend_f();\n'
+        return (f'void (*frontend_{uuid})(void) = {sig};\n'
+                f'frontend_{uuid}();\n')
+
+    def _format_indirect_call_multitarget(self, branch: blocks.Branch,
+                                          uuid: int) -> str:
+        paths = branch.next_target_sequence()
+        max_index = len(paths)
+        paths_array_name = f'paths_{uuid}'
+        paths_array = self._create_path_sequence_array(paths, paths_array_name)
+        func_array = self._create_call_target_array(branch.get_targets(),
+                                                    f'array_{uuid}')
+        result = (
+            f'static int index_{uuid} = 0;\n'
```

Comment:
Example output of two target indirect call: 

```
static int index_139664210237616 = 0;
static int paths_139664210237616[16] = {1,1,1,0,1,1,1,0,0,0,1,1,1,0,1,0};
static void* array_139664210237616[] = {&function_3, &function_4};
void (*f_139664210237616)(void) = array_139664210237616[paths_139664210237616[index_139664210237616++ % 16]];
f_139664210237616();
```

Reply:
Cool!